javascript.js

```diff
@@ -0,0 +1,146 @@
+};
+
+function displayRecipe(recipe) {
+  const container = document.getElementById("recipe-container");
+
+  const recipeElement = document.createElement("div");
+  recipeElement.className = "recipe";
+
+  const title = document.createElement("h2");
+  title.textContent = recipe.name;
+  recipeElement.appendChild(title);
+
+  const description = document.createElement("p");
+  description.textContent = recipe.description;
+  recipeElement.appendChild(description);
+
+  const ingredientsTitle = document.createElement("h3");
+  ingredientsTitle.textContent = "Ingredients:";
+  recipeElement.appendChild(ingredientsTitle);
+
+  const ingredientsList = document.createElement("ul");
+  recipe.ingredients.forEach((ingredient) => {
+    const listItem = document.createElement("li");
+    listItem.textContent = ingredient;
+    ingredientsList.appendChild(listItem);
+  });
+  recipeElement.appendChild(ingredientsList);
+
+  container.appendChild(recipeElement);
+}
+displayRecipe(recipe);
+
+let ingredientCount = 5;
+
+document
+  .getElementById("recipe-form")
+  .addEventListener("submit", function (event) {
+    event.preventDefault();
+
+    const name = document.getElementById("name").value;
+    const description = document.getElementById("description").value;
+    const pictureUrl = document.getElementById("pictureUrl").value;
+    const ingredients = [];
+
+    for (let i = 1; i <= ingredientCount; i++) {
+      const ingredient = document.getElementById(`ingredient${i}`).value;
+      if (ingredient) {
+        ingredients.push(ingredient);
+      }
+    }
+
+    const newRecipe = {
+      id: Date.now(),
+      name,
+      description,
+      ingredients,
+      pictureUrl,
+    };
+
+    displayRecipe(newRecipe);
+    document.getElementById("recipe-form").reset();
+    resetIngredientFields();
+  });
+
+function addIngredient() {
+  ingredientCount++;
+
+  const newIngredient = document.createElement("div");
+  newIngredient.innerHTML = `
+    <label for="ingredient${ingredientCount}">Ingredient ${ingredientCount}:</label>
+    <input type="text" id="ingredient${ingredientCount}" name="ingredient${ingredientCount}"><br><br>
+  `;
+
+  document.getElementById("ingredients-container").appendChild(newIngredient);
+}
+
+function resetIngredientFields() {
+  ingredientCount = 5;
+  document.getElementById("ingredients-container").innerHTML = `
```

Comment:
You could only reset the inputs instead of re-implementing the form. Also, this section is a good candidate for using a loop.